internal/k8s/controllers/layer2_status_controller.go

```diff
@@ -85,8 +90,16 @@ func (r *Layer2StatusReconciler) Reconcile(ctx context.Context, req ctrl.Request
 
 	var result controllerutil.OperationResult
 	result, err = controllerutil.CreateOrPatch(ctx, r.Client, state, func() error {
-		state.Labels = map[string]string{LabelAnnounceNode: r.NodeName, LabelServiceName: svcName}
+		state.Labels = map[string]string{
+			LabelAnnounceNode:     r.NodeName,
+			LabelServiceName:      req.Name,
+			LabelServiceNamespace: req.Namespace,
+		}
 		state.Status = desiredStatus
+		err = controllerutil.SetOwnerReference(r.SpeakerPod, state, r.Scheme())
+		if err != nil {
+			level.Warn(r.Logger).Log("error setting owner ref", err)
```

Comment:
I was considering maintaining owner reference as an extra function in addition to the core function, rather than a part of the core function.
So I didn't do a fast fail when facing errors about owner reference.